### ISSUE TITLE
Unable to use services requiring an API Login

### ISSUE THREAD

**Issue by anakaine** [openaddresses/pyesridump#47]
Where an API login page is presented the esridump tool ceases to function. 
There are cases where a service is accessed using legitimate credentials, but the esridump tool does not have a facility for authentication such as either username/password, or token.

**Comment by iandees** [comment]
Thanks for reporting this. From what I've been able to discern, the login process gives you an API token that can then be used by the tool as-is. I've never had to use a login though, so I'm not sure how it should work. I'd appreciate a pull request or an example layer that requires a password.

**Comment by anakaine** [comment]
I'll have a sample service for you soon.

**Comment by iandees** [comment]
I think I'd prefer to keep login functionality out of pyesridump. You can use the server's "generateToken" endpoint (`/arcgis/tokens/generateToken`) to create an API token and then pass that in to `esri2geojson` command line or in the `extra_query_args` param in the `EsriDumper` object.

The login process requires that the user pass in a new URL to esridump system (the URL to generate tokens can be different and isn't deterministic).

I think I'll update the documentation/readme to point that process out.

**Comment by marhensa** [comment]
any update of documentation to use mapservice with a token?

**Comment by anakaine** [comment]
4 years since the last discussion, and I've not looked at the process
since, but: a mapservice generally returns either images, or a feature at a
point whose coordinated have been queried. It does not return a list of
features which can then be queried by their IDs, which is what the script
was doing last I looked.

Locking stuff up in a map service makes extraction difficult.

**Comment by iandees** [comment]
> something like this:
> 
> esri2geojson -p token=xxxxlongstringsoftokensxxx https://some.site/server/rest/services/folders/folder/MapServer/0 output.geojson

Yep, pass the token you retrieved through other means as a URL parameter using the `-p` argument to esri2geojson.
